frontend/src/templates/Field/ChildrenCompound/ChildrenCompoundField.tsx

```diff
@@ -322,33 +315,28 @@ const ChildrenBody = ({
         <FormLabel gridArea="formlabel">Child</FormLabel>
         <Flex align="stretch" alignItems="stretch" justify="space-between">
           <Box flexGrow={10}>
-            <FormControl key={field.id} isRequired isInvalid={!!childNameError}>
+            <FormControl key={field.id} isRequired isInvalid={!!error?.[0]}>
```

Comment:
Sudden use of hardcoded index access (i.e., `[0]`) looks suspicious. Should it be the `index` of the current child instead?

i.e., Child Component has two rows, the validation error is on 2nd child instead, would this `error?.[0]` pick up the error on the correct child?

Reply:
hi ken, seems like the hardcoding of the index where 0th element refers to the name field is a deterministic behaviour since a while ago. (see attached ss) 

<img width="702" alt="image" src="https://github.com/user-attachments/assets/0638f6df-a226-4675-a3a2-db022e0afdd3">

Reply:
what i can do is define a constant to document this behavior so its clearer.

Reply:
Yeah I think `const NAME_INDEX = 0` gives it more clarity, since I was also confused why it is always the first index. Explicitly calling it out makes sense.

Reply:
ill add it in then